e2e/app/workspace-card.ts

```diff
@@ -41,24 +42,25 @@ export default class WorkspaceCard extends BaseElement {
   }
 
   static async findCard(page: Page, workspaceName: string): Promise<WorkspaceCard | null> {
-    const cards = await page.$x(WorkspaceCard.cardRootXpath);
-    for (const card of cards) {
-      const children = await card.$x(`.//*[@data-test-id="workspace-card-name" and text()="${workspaceName}"]`);
+    const allCards = await this.getAllCards(page);
+    for (const card of allCards) {
+      const children = await card.asElementHandle().$x(`.//*[@data-test-id="workspace-card-name" and text()="${workspaceName}"]`);
       if (children.length > 0) {
-        return new WorkspaceCard(page).asCardElement(card);
+        return card; // matched workspace name, found the Workspace card.
       }
-      await card.dispose();
+      await card.dispose(); // not it, dispose the ElementHandle.
     }
-    return null;
+    return null; // not found
```

Comment:
Calling `getAllCards()` function to avoid duplicate code. should have been done at the beginning.